### PR TITLE
Create com.github.donadigo.appeditor

### PR DESCRIPTION
**Pull request by bil-elmoussaoui**

### CODE REVIEW

File: com.github.donadigo.appeditor [hunk 1]
```diff
@@ -0,0 +1,81 @@
+{
+    "app-id": "com.github.donadigo.appeditor",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "command": "com.github.donadigo.appeditor",
+    "build-options": {
+        "cflags": "-O2",
+        "cxxflags": "-O2"
```
Comment (TingPing) [inline com.github.donadigo.appeditor]: These shouldn't be needed, setting the build types on the projects sets compiler flags.

File: com.github.donadigo.appeditor [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    },
+    "cleanup": [
+        "/bin/granite-demo",
```
Comment (TingPing) [inline com.github.donadigo.appeditor]: Just personal preference but I like cleanups like this one to be in the module they came from.

File: com.github.donadigo.appeditor (hunk 1 above)
Comment (bil-elmoussaoui) [reply com.github.donadigo.appeditor]: Done

**Comment by TingPing** [comment]
bot, build com.github.donadigo.appeditor

File: com.github.donadigo.appeditor (hunk 2 above)
Comment (bil-elmoussaoui) [reply com.github.donadigo.appeditor]: I have moved the granite specific files to the granite module

**Comment by TingPing** [comment]
Filename missing `.json` extension.

**Comment by bil-elmoussaoui** [comment]
already fixed

File: com.github.donadigo.appeditor.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        "*.a",
+        "*.la"
+    ],
+    "finish-args": [
+        /* X11 + XShm */
+        "--share=ipc",
+        "--socket=x11",
+        /* Wayland */
+        "--socket=wayland",
+        "--filesystem=host:ro",
```
Comment (TingPing) [inline com.github.donadigo.appeditor.json]: You'll also want: `--filesystem=~/.local/share/applications:create`

Comment (bil-elmoussaoui) [reply com.github.donadigo.appeditor.json]: Done

**Comment by bil-elmoussaoui** [comment]
@TingPing Fixed everything. Still somehow Glib can't read any desktop file, and can't neither report which if a binary file does exists (using find_program)

File: com.github.donadigo.appeditor.json (hunk 3 above)
Comment (TingPing) [reply com.github.donadigo.appeditor.json]: Sorry I did update my comment to be `xdg-data` rather than `~/.local/share`

**Comment by TingPing** [comment]
```
07:48:06| <TingPing> belmoussaoui, like i said, its in /run/host
07:48:19| <TingPing> belmoussaoui, the app will need to be taught to look there
07:48:23| <belmoussaoui> yeah, the application uses Glib to get the list of user & system data dirs 
07:48:41| <TingPing> you can set XDG_DATA_DIRS, which probably won't explode
08:01:16| <TingPing> belmoussaoui, you'll need a wrapper script for that btw if not clear
08:01:59| <TingPing> just runs env XDG_DATA_DIRS=$XDG_DATA_DIRS:/run/host/usr/share com.github.donadigo.appeditor
```

**Comment by bil-elmoussaoui** [comment]
@TingPing I have added the wrapper, things works correctly now. It doesn't detect every desktop file, but that must be an upstream issue I guess...

**Comment by nedrichards** [comment]
bot, build com.github.donadigo.appeditor

**Comment by bil-elmoussaoui** [comment]
The build somehow wasn't triggered at all...

**Comment by nedrichards** [comment]
bot, build com.github.donadigo.appeditor

**Comment by nedrichards** [comment]
bot, build com.github.donadigo.appeditor

**Comment by nedrichards** [comment]
@bil-elmoussaoui have you proposed the patch upstream? Or a more long term fix?

**Comment by nedrichards** [comment]
I ran a test. It didn't find the exported system desktop files or those from flatpaks, may just be an Endless OS issue, but still, not ideal.

The 'open in text editor' button on an entry doesn't seem to work on my system either. Not sure what command it uses to run an app.

The app is somewhat visually degraded without the Elementary theme and icons.

**Comment by bil-elmoussaoui** [comment]
@nedrichards The patch is just an issue with the latest versions of GLib. Still waiting for upstream to provide a better solution (with a condition to check the version of GLib).

The application works and looks okay for me in GNOME. The Open in text editor option works here too

**Comment by nedrichards** [comment]
OK - lgtm

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.donadigo.appeditor

**Pull request closed without merging**